nifpga/session.py

```diff
@@ -390,8 +410,272 @@ def read(self):
+        if self._overflow_enabled:
+            self.overflow = self._get_overflow_value(data)
+            data = self._remove_overflow_bit(data)
+        if self._signed:
+            data = self._integer_twos_comp(data)
+        return Decimal(data * self._delta)
+
+    def _get_overflow_value(self, data):
+        """ Mask out all the data within the word length, leaving the overflow
+        bit. If the result after masking the the word portion of the fixed
+        point is none zero that indicates the data read has overflown."""
+        mask = 2**(self._word_length)
+        if data & mask > 0:
+            return True
+        return False
+
+    def _remove_overflow_bit(self, data):
+        """ This helper function masks out all bits not inside the word length,
+        ultimately returning a value of data missing the overflow bit. """
+        return data & (2**(self._word_length) - 1)
+
+    def _integer_twos_comp(self, data):
+        if (data & (1 << (self._word_length - 1))) != 0:
+            data = data - (1 << self._word_length)
+        return data
+
+    def write(self, data):
+        """ Converts the passed in argument into fixed point representation
+        and then writes it into the register.
+
+            Args:
+                data (any python datatype): The data to be written into the
+                register. This method will coerce the value to the nearest
+                possible fixed point representation.
+
+                self.overflow: Registers with overflow enabled will write the
+                value of the attribute into the register. Users should set this
+                attribute on the the register before calling write.
+        """
+        binary = self._convert_data_to_binary_fxp(data)
+        arrayData = []
+        for index in range(0, len(self)):
+            blocksize = 32 * index
+            arrayData.append(int(binary[0 + blocksize: 32 + blocksize], 2))
+        super(_FxpRegister, self).write(arrayData)
+
+    def _convert_data_to_binary_fxp(self, data):
+        """ This function will convert any python datatype (decimal, integer,
+         float) into a binary that can be written to the RIO device the same
+        way a LabVIEW Host interface would."""
+        binary = ''
+        if not self._signed and data < 0:
+            binary = ''.zfill(self._word_length)
+            warn_coerced_data()
+        elif self._integer_word_length >= self._word_length:
+            binary = self._calculate_binary_integer_from_data(data)
+            binary = binary[:self._word_length]
+        elif self._integer_word_length < 0:
+            binary = self._calculate_binary_fraction_from_data(data)
+            binary = binary[len(binary) - self._word_length:]
+        else:
+            interger_part = self._calculate_binary_integer_from_data(data)
+            fractional_part = self._calculate_binary_fraction_from_data(data)
+            binary = interger_part + fractional_part
```

Comment:
I wasn't expecting this piecemeal conversion. I thought I would see something like:
Normalize(val): return val / FxpDelta
Normalize should shift the significant bits to the integer position, then you can truncate extra bits at the top and bottom of the word and report errors.

Reply:
Removed in "Removing string based implementation commit"